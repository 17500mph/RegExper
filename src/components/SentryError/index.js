import React from 'react';
import PropTypes from 'prop-types';
import * as Sentry from '@sentry/browser';
import { withTranslation, Trans } from 'react-i18next';

import Message from 'components/Message';

const reportError = event => {
  event.preventDefault();

  if (Sentry.lastEventId()) {
    Sentry.showReportDialog();
  }
};

export const SentryError = ({ t }) => (
  <Message type="error" heading={ t('An error has occurred') }>
    <p>
      <Trans>This error has been logged. You may also <a href="#error-report"
        onClick={ reportError }>fill out a report</a>.</Trans>
    </p>
  </Message>
);

SentryError.propTypes = {
  t: PropTypes.func.isRequired
};

export default withTranslation()(SentryError);
