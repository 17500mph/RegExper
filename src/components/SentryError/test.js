jest.mock('@sentry/browser');

import React from 'react';
import { shallow } from 'enzyme';
import * as Sentry from '@sentry/browser';

import SentryError from 'components/SentryError';

describe('SentryError', () => {
  test('rendering', () => {
    const component = shallow(
      <SentryError />
    );
    expect(component).toMatchSnapshot();
  });

  describe('error reporting', () => {
    test('fill out a report when an event has been logged', () => {
      Sentry.lastEventId.mockReturnValue(1);
      const component = shallow(
        <SentryError />
      );
      const eventObj = { preventDefault: jest.fn() };
      component.find('a').simulate('click', eventObj);

      expect(eventObj.preventDefault).toHaveBeenCalled();
      expect(Sentry.showReportDialog).toHaveBeenCalled();
    });

    test('fill out a report when an event has not been logged', () => {
      Sentry.lastEventId.mockReturnValue(false);
      const component = shallow(
        <SentryError />
      );
      const eventObj = { preventDefault: jest.fn() };
      component.find('a').simulate('click', eventObj);

      expect(eventObj.preventDefault).toHaveBeenCalled();
      expect(Sentry.showReportDialog).not.toHaveBeenCalled();
    });
  });
});
