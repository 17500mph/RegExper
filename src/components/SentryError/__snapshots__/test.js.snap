// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SentryError rendering 1`] = `
<Message
  heading="TRANSLATE(An error has occurred)"
  type="error"
>
  <p>
    <Trans>
      This error has been logged. You may also 
      <a
        href="#error-report"
        onClick={[Function]}
      >
        fill out a report
      </a>
      .
    </Trans>
  </p>
</Message>
`;
